### PR TITLE
Add test to validate 'Assembly.LoadFrom' behavior

### PR DESCRIPTION
**Pull request by daxian-dbw**
Add validation test for #4192

### CODE REVIEW

File: test/powershell/engine/Basic/Assembly.LoadFrom.ps1 [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                {
+                    public static string GetProviderName()
+                    {
+                        return "Assembly.Bar.Provider";
+                    }
+                }
+            }
+'@
+
+        $TempPath = [System.IO.Path]::GetTempFileName()
```
Comment (SteveL-MSFT) [inline test/powershell/engine/Basic/Assembly.LoadFrom.ps1]: Why not just use $testdrive?

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: test/powershell/engine/Basic/Assembly.LoadFrom.ps1 (hunk 1 above)
Comment (daxian-dbw) [inline test/powershell/engine/Basic/Assembly.LoadFrom.ps1]: Once the assembly is loaded, the file cannot be removed. If I use `$testdrive`, Pester will try to remove it afterward and result in non-failure errors.

Comment (SteveL-MSFT) [inline test/powershell/engine/Basic/Assembly.LoadFrom.ps1]: Got it, perhaps add a comment

Comment (daxian-dbw) [inline test/powershell/engine/Basic/Assembly.LoadFrom.ps1]: A comment is added.

**Review by SteveL-MSFT — APPROVED** [review]
LGTM

**Comment by iSazonov** [comment]
Should we test loading from different default paths? It seems we had an issue with CoreFX.

**Comment by daxian-dbw** [comment]
@iSazonov can you please elaborate? And which CoreFx issue?

**Comment by iSazonov** [comment]
I mean the comment https://github.com/PowerShell/PowerShell/issues/4192#issuecomment-313457476

**Comment by daxian-dbw** [comment]
This test is to cover the corefx issue described in that comment. `Consumer.dll` and `Provider.dll` are in the same folder and the former implicit references to the latter.

**Pull request merged**